src/consumeroffset/kafka_offset_collect.go

```diff
@@ -231,3 +253,112 @@ func collectInactiveConsumerGroupOffsets(
+		clusterIDAttr := integration.NewIDAttribute("clusterName", args.GlobalArgs.ClusterName)
+
+		consumerGroupEntity, err := kafkaIntegration.Entity(string(consumerGroup), nrConsumerGroupEntity, clusterIDAttr)
+		if err != nil {
+			log.Error("Failed to get entity for consumer group: %s", err)
+			continue
+		}
+
+		ms := consumerGroupEntity.NewMetricSet("KafkaOffsetSample",
+			attribute.Attribute{Key: "clusterName", Value: args.GlobalArgs.ClusterName},
+			attribute.Attribute{Key: "consumerGroup", Value: string(consumerGroup)},
+			attribute.Attribute{Key: "clusterName", Value: args.GlobalArgs.ClusterName},
+		)
+
+		err = ms.SetMetric("consumerGroup.totalLag", totalLag, metric.GAUGE)
+		if err != nil {
+			log.Error("Failed to set metric consumerGroup.totalLag: %s", err)
+		}
+
+		maxLag := consumerGroupMaxLagRollup[consumerGroup]
+		err = ms.SetMetric("consumerGroup.maxLag", maxLag, metric.GAUGE)
+		if err != nil {
+			log.Error("Failed to set metric consumerGroup.maxLag: %s", err)
+		}
+
+		err = ms.SetMetric("consumerGroup.activeConsumers", len(cGroupActiveClientsRollup), metric.GAUGE)
+		if err != nil {
+			log.Error("Failed to set metric consumerGroup.activeConsumers: %s", err)
+		}
+	}
+}
+
+func consumerGroupByTopicMetrics(
+	topicRollup map[topic]int,
+	consumerGroup string,
+	kafkaIntegration *integration.Integration,
+	topicMaxLagRollup map[topic]int,
+	topicActiveClientsRollup map[topic]map[clientID]struct{},
+) {
+	for topic, totalLag := range topicRollup {
+		clusterIDAttr := integration.NewIDAttribute("clusterName", args.GlobalArgs.ClusterName)
+		consumerGroupIDAttr := integration.NewIDAttribute("consumerGroup", consumerGroup)
+		topicIDAttr := integration.NewIDAttribute("topic", string(topic))
+
+		partitionConsumerEntity, err := kafkaIntegration.Entity(string(topic), nrConsumerGroupTopicEntity, clusterIDAttr, consumerGroupIDAttr, topicIDAttr)
+		if err != nil {
+			log.Error("Failed to get entity for partition consumer: %s in topic %s", err, topic)
+			return
+		}
+
+		ms := partitionConsumerEntity.NewMetricSet("KafkaOffsetSample",
+			attribute.Attribute{Key: "clusterName", Value: args.GlobalArgs.ClusterName},
+			attribute.Attribute{Key: "consumerGroup", Value: consumerGroup},
+			attribute.Attribute{Key: "topic", Value: string(topic)},
+		)
+
+		err = ms.SetMetric("consumerGroup.totalLag", totalLag, metric.GAUGE)
+		if err != nil {
+			log.Error("Failed to set metric consumerGroup.totalLag: %s for topic: %s", err, topic)
+		}
+
+		maxLag := topicMaxLagRollup[topic]
+		err = ms.SetMetric("consumerGroup.maxLag", maxLag, metric.GAUGE)
+		if err != nil {
+			log.Error("Failed to set metric consumerGroup.maxLag: %s for topic: %s", err, topic)
+		}
+
+		err = ms.SetMetric("consumerGroup.activeConsumers", len(topicActiveClientsRollup[topic]), metric.GAUGE)
+		if err != nil {
+			log.Error("Failed to set metric consumerGroup.activeConsumers: %s for topic: %s", err, topic)
+		}
+	}
+}
```

Comment:
Would it make sense to create specific unit tests for the functions that get NR metrics from aggregations? After all those functions are helpers so I am not sure, maybe they are already properly covered with the existent tests.

Reply:
They are covered in the actual tests